packages/odyssey-react-mui/src/Dialog.tsx

```diff
@@ -0,0 +1,71 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { Dialog as MuiDialog } from "@mui/material";
+import { DialogProps as MuiDialogProps } from "@mui/material";
+import {
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+  DialogActions,
+} from "@mui/material";
+import { Button, CloseIcon } from "./";
+import { memo, ReactNode, useState, useEffect, useRef } from "react";
+
+export interface DialogProps extends MuiDialogProps {
+  onClose: () => void;
+  title: string;
+  actions?: ReactNode;
+}
```

Comment:
I wanna specify our own props rather than extending MUI.

For example, if I said "what props does this support?" Could you answer or would you have to go to MUI's docs? If MUI's docs, then we don't control the component API.

That's why I say it. I want all Odyssey folks to understand what we support at-a-glance.